DigitalLearningSolutions.Data.Tests/DataServices/SystemNotificationsDataServiceTests.cs

```diff
@@ -0,0 +1,115 @@
+﻿namespace DigitalLearningSolutions.Data.Tests.DataServices
+{
+    using System;
+    using System.Linq;
+    using System.Transactions;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Models;
+    using DigitalLearningSolutions.Data.Tests.TestHelpers;
+    using FluentAssertions;
+    using FluentAssertions.Execution;
+    using NUnit.Framework;
+
+    public class SystemNotificationsDataServiceTests
+    {
+        private SystemNotificationsDataService systemNotificationsDataService = null!;
+        private SystemNotificationTestHelper systemNotificationTestHelper = null!;
+
+        [SetUp]
+        public void Setup()
+        {
+            var connection = ServiceTestHelper.GetDatabaseConnection();
+            systemNotificationsDataService = new SystemNotificationsDataService(connection);
+            systemNotificationTestHelper = new SystemNotificationTestHelper(connection);
+        }
+
+        [Test]
+        public void AcknowledgeSystemNotification_creates_new_record()
+        {
+            using var transaction = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+            try
+            {
+                // Given
+                const int notificationId = 1;
+                const int adminId = 5;
+
+                // When
+                systemNotificationsDataService.AcknowledgeNotification(notificationId, adminId);
+                var acknowledgedNotificationIds =
+                    systemNotificationTestHelper.GetSystemNotificationAcknowledgementsForAdmin(adminId);
+
+                // Then
+                acknowledgedNotificationIds.Should().Contain(notificationId);
+            }
+            finally
+            {
+                transaction.Dispose();
+            }
+        }
+
+        [Test]
+        public void GetUnacknowledgedSystemNotifications_returns_correct_notifications()
+        {
+            using var transaction = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+            try
+            {
+                // Given
+                const int adminId = 5;
+                var currentTime = DateTime.UtcNow.Date;

```

Comment:
I know they are all part of the given, but can you pull all this stuff that can't throw an exception and isn't part of the transaction out before them so we have a shorter try block